compiler/damlc/daml-prim-src/GHC/Base.daml

```diff
@@ -87,3 +89,11 @@ class Functor f where
     (<$)        =  fmap . const
 
 infixl 4  <$
+
+infixr 0 $
+-- | Take a function from `a` to `b` and a value of type `a`, and apply the
+-- function to the value of type `a`, returning a value of type `b`.
+-- This function has a very low precedence, which is why you might want to use
+-- it instead of regular function application.
+($) : forall r a (b : TYPE r). (a -> b) -> a -> b
```

Comment:
Let's hope nobody has used `$` with explicit type applications so far. I mean, why would you, but still. And there's a simple workaround: just don't use `$`. 💸

Reply:
So, technically this is breaking the existing standard library interface. But it's also a bug fix in the sense that it makes `($)` work with ApplicativeDo. Do you want to weigh in on this, @bame-da?

Reply:
All good from my end.